src/components/Identity/getIdentity/getIdentityOptionsValidator.js

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { objectOf, literal, arrayOf } from "../../../utils/validation";
+/**
+ * Verifies user provided event options.
+ * @param {*} options The user event options to validate
+ * @returns {*} Validated options
+ */
+export default options => {
+  const getIdentityOptionsValidator = objectOf({
+    namespaces: arrayOf(literal("ECID")).nonEmpty()
+  }).noUnknownFields();
+  getIdentityOptionsValidator(options);
+  // Return default options for now
+  // To-Do: Accept namespace from given options
+  return {
+    namespaces: ["ECID"]
+  };
```

Comment:
I think this validator can be approved. Rather than returning an object like this, you should be able to do it all in the validator itself. Something like this, I think (I haven't tested it):

```
export default objectOf({
    namespaces: arrayOf(literal("ECID")).nonEmpty().unique().required()
  }).noUnknownFields().default({
    namespaces: ["ECID"]
  });
```

@jonsnyder might have some tips here.

Reply:
Couldn't do that because `default` wasn't working for objects.
default does a null check but empty object returns true

Reply:
This sounds like something that should be looked into further then. I'll dig into it and see what I can find.

Reply:
It's going to need some work before we can introduce the change. I logged an issue with my findings here: https://jira.corp.adobe.com/browse/CORE-45013

Reply:
I also added a JIRA issue to created a validator that would allow you to validate that each namespace is unique: https://jira.corp.adobe.com/browse/CORE-45016

Reply:
Thanks, @Aaronius, for logging these issues.